File: app/src/main/java/com/udeshcoffee/android/ui/dialogs/DeleteEQDialog.java
package com.udeshcoffee.android.ui.dialogs;

import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;

import com.udeshcoffee.android.ui.main.equalizer.EqualizerFragment;

/**
 * Created by Udathari on 2/25/2017.
 */

public class DeleteEQDialog extends DialogFragment {

    public static String ARGUMENT_PRESET = "ARGUMENT_PRESET";
    public static String ARGUMENT_NAME = "ARGUMENT_NAME";

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        final int preset = this.getArguments().getInt(ARGUMENT_PRESET);
        final String name = this.getArguments().getString(ARGUMENT_NAME);
        AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
        builder.setTitle("Delete")
                .setMessage("Delete " + name + " preset?")
                // Add action buttons
                .setPositiveButton("Delete", new DialogInterface.OnClickListener(){

                            @Override
                            public void onClick(DialogInterface dialogInterface, int i) {
                                if (getTargetFragment() != null) {
                                    ((EqualizerFragment) getTargetFragment()).onDeleteEQDialogResult(preset);
                                }
                            }
                        }
                )
                .setNegativeButton("Cancel", new DialogInterface.OnClickListener(){

                    @Override
                    public void onClick(DialogInterface dialogInterface, int i) {
                        dismiss();
                    }
                });
        return builder.create();

    }

}

File: app/src/main/java/com/udeshcoffee/android/ui/common/dialogs/DeleteEQDialog.kt
package com.udeshcoffee.android.ui.common.dialogs

import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment
import androidx.appcompat.app.AlertDialog
import com.udeshcoffee.android.ui.main.equalizer.EqualizerFragment

/**
* Created by udesh on 3/21/18.
*/

class DeleteEQDialog : androidx.fragment.app.DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val preset = this.arguments!!.getInt(ARGUMENT_PRESET)
        val name = this.arguments!!.getString(ARGUMENT_NAME)
        val builder = AlertDialog.Builder(context!!)
        builder.setTitle("Delete")
                .setMessage("Delete $name preset?")
                // Add action buttons
                .setPositiveButton("Delete"
                ) { _, _ ->
                    if (targetFragment != null) {
                        (targetFragment as EqualizerFragment).onDeleteEQDialogResult(preset)
                    }
                }
                .setNegativeButton("Cancel") { _, _ -> dismiss() }
        return builder.create()

    }

    companion object {
        var ARGUMENT_PRESET = "ARGUMENT_PRESET"
        var ARGUMENT_NAME = "ARGUMENT_NAME"
        fun create(preset: Int, name: String): DeleteEQDialog {
            val mDialog = DeleteEQDialog()
            val bundle = Bundle()
            bundle.putInt(DeleteEQDialog.ARGUMENT_PRESET, preset)
            bundle.putString(DeleteEQDialog.ARGUMENT_NAME, name)
            mDialog.arguments = bundle
            return mDialog
        }
    }

}
